rust/src/statistics/formatters/writer_with_indentation.rs

```diff
@@ -0,0 +1,325 @@
+// Copyright (c) 2025 Sonic Operations Ltd
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at soniclabs.com/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+use std::io::Write;
+
+use crate::statistics::{
+    Statistic,
+    formatters::StatisticsFormatter,
+    node_count::{NodeCountStatistic, NodeCountsByKindStatistic, NodeCountsByLevelStatistic},
+};
+
+/// A statistics formatter that writes statistics to a writer with indentation support.
+pub struct WriterWithIndentation<W: Write> {
```

Comment:
```suggestion
pub struct HumanReadableWriter<W: Write> {
```

I think the indentation is more of an "implementation detail", the point is that this is human-readable formatting. WDYT?